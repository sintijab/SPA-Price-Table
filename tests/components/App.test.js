import React from 'react';
import App from '../../js/components/View';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import * as axios from "axios";
jest.mock("axios");

Enzyme.configure({ adapter: new Adapter() });

describe('<App />', () => {
  const props = {
    dataFailed: () => {},
    dataLoaded: () => {},
    dataEmpty: () => {},
    modalState: [{
      id: 7,
      name: "InsuranceFirma",
      budget: 9787866,
      budget_spent: 46000,
      date_of_first_purchase: "2409-07-07"
    }],
  };

  it('renders <App /> component', () => {
    const wrapper = shallow(<App item={props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('calls getFinanceData', () => {
    const wrapper = shallow(<App item={props} />);
    const spy = jest.spyOn(wrapper.instance(), 'getFinanceData');
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });
});
