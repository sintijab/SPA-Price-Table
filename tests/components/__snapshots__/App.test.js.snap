// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> calls getFinanceData 1`] = `
<div>
  <Connect(ViewContainer) />
</div>
`;

exports[`<App /> renders <App /> component 1`] = `
<div>
  <Connect(ViewContainer) />
</div>
`;
